.github/workflows/Tests.yaml

```diff
@@ -0,0 +1,227 @@
+
+  # Build the documentation and check for problems.
+  doc_check:
+    name: Documentation
+    # No need to do this again when opening a PR, so only check on `push`
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    container:
+      image: sxscollaboration/spectrebuildenv:latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Configure with cmake
+        working-directory: /work
+        run: >
+          mkdir build && cd build
+
+          cmake
+          -D CMAKE_C_COMPILER=clang
+          -D CMAKE_CXX_COMPILER=clang++
+          -D CMAKE_Fortran_COMPILER=gfortran-8
+          -D CHARM_ROOT=/work/charm/multicore-linux64-clang
+          -D CMAKE_BUILD_TYPE=Debug
+          -D DEBUG_SYMBOLS=OFF
+          -D BUILD_PYTHON_BINDINGS=ON
+          $GITHUB_WORKSPACE
+      - name: Check documentation
+        working-directory: /work/build
+        # Append `texlive` to the `PATH` so bibliography generation works.
+        # Can be removed once this is done in the container.
+        run: |
+          PATH=$PATH:/work/texlive/bin/x86_64-linux make doc-check
+      # Archive the built documentation so it can be downloaded and reviewed
+      - name: Archive documentation
+        uses: actions/upload-artifact@v1
+        with:
+          name: docs-html
+          path: /work/build/docs/html
+
+  # Build all test executables and run unit tests on a variety of compiler
+  # configurations.
+  unit_tests:
+    name: Unit tests
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        compiler: [gcc-6, gcc-7, gcc-8, gcc-9, clang-4.0, clang-5.0, clang-8]
+        build_type: [Debug, Release]
+    container:
+      image: sxscollaboration/spectrebuildenv:latest
+      env:
+        # We make sure to use a fixed absolute path for the ccache directory
+        CCACHE_DIR: /work/ccache
+        # GitHub Actions currently limits the size of individual caches
+        # to 400 MB.
+        CCACHE_MAXSIZE: 400M
+        CCACHE_COMPRESS: 1
+        CCACHE_COMPRESSLEVEL: 6
+        # In order to make ccache work with precompiled headers we need to:
+        # - Hash the content of the compiler rather than the location and mtime
+        # - Hash the header file in the repo that will generate the precompiled
+        #   header
+        # - Ignore the precompiled header in the build directory
+        CCACHE_COMPILERCHECK: content
+        CCACHE_EXTRAFILES: ${GITHUB_WORKSPACE}/tools/SpectrePch.hpp
+        CCACHE_IGNOREHEADERS:
+          /work/build/SpectrePch.hpp:/work/build/SpectrePch.hpp.gch
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Restore ccache
+        uses: actions/cache@v1
+        with:
+          path: /work/ccache
+          key: ccache-${{ matrix.compiler }}-${{ matrix.build_type }}
+          restore-keys: |
+            ccache-${{ matrix.compiler }}
```

Comment:
These `restore-keys` are fallbacks for when the `key` is not found. Please see [here](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/caching-dependencies-to-speed-up-workflows#matching-a-cache-key) in the docs. They are only useful if `ccache` can make any use of a cache that was generated by another build type or compiler(-version).. do you know if it can?

Reply:
Ah okay! No, it cannot because you get a different cache for different compiler flags, versions, etc.

Reply:
Ok, removed the `restore-keys`.